scanpy/tests/test_clustering.py

```diff
@@ -19,6 +20,50 @@ def adata_neighbors():
 def test_leiden_basic(adata_neighbors):
     sc.tl.leiden(adata_neighbors)
 
+@pytest.mark.parametrize('clustering,key', [
+    (sc.tl.louvain, 'louvain'),
+    (sc.tl.leiden, 'leiden'),
+])
+def test_clustering_subset(adata_neighbors, clustering, key):
+    clustering(adata_neighbors, key_added=key)
+    print(adata_neighbors)
```

Comment:
If an assertion for a test with a fixture fails, pytest will print the value of the passed fixture by itself.

For example:

```python
import pytest

@pytest.fixture
def foo():
    return "foo"

def test_foo(foo):
    assert False
```

Fails with a message like:

```python
============================================= FAILURES ==============================================
_____________________________________________ test_foo ______________________________________________

foo = 'foo'

    def test_foo(foo):
>       assert False
E       assert False

tests/test_module.py:8: AssertionError
```

Similar for parameterize.

Reply:
Since `adata_neighbors` is a fixture, I think it'll get printed when the test fails.

Reply:
Sorry, I thought I had corrected it. It should be fixed now.

Reply:
No worries, thanks!